source/use-cases/sensitive-data-encryption.txt

```diff
@@ -300,6 +289,15 @@ information:
 - The BSON Type of each field (only required for deterministically
   encrypted fields)
 
+.. note::
+   
+   While a MongoDB client can use JSON Schema for validating types and values
+   of a document, the encrypted client can only process instructions for
+   automatic client-side encryption and decryption. In this example, we define
+   a JSON Schema for the encrypted client only. For information on type 
+   and value validation with JSON Schema, see 
+   :manual:`Schema Validation </core/schema-validation/>`.
```

Comment:
AFAIK clients cannot use JSON Schema for validation at all. It's a feature that exclusively lives on the server.

```rst
.. admonition:: JSON Schema Validation
   :important:
   
   MongoDB drivers only use JSON schema to configure automatic
   client-side field level encryption. If you specify a document
   validation keyword in your CSFLE JSON Schema then the client will
   throw an error (when?).
   
   For information on how to use JSON Schema on the server to validate
   data, see :manual:`Schema Validation </core/schema-validation/>`.
```

- When does the client error (on creation? on query? etc.)

Reply:
Thanks. Do you think we may need a more detailed explanation of "document validation keyword" for users that may not be familiar with the server-based validation? Will investigate the Java driver to see when the error is thrown and check if it exists in the driver specification.

Reply:
According to the driver specification it should error, so the Java driver is likely not up to spec, or the spec is outdated:

From: https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#schemamap

> Schemas supplied in the schemaMap only apply to configuring automatic encryption for client side encryption. Other validation rules in the JSON schema will not be enforced by the driver and will result in an error.

Reply:
Hmm the spec is also pretty unclear ("will result in an error _when????_") so maybe we just leave it out for now. Up to you - thanks for checking!

Reply:
My reading is that specifying validation rules in the JSON schema should *always* error. This is also my understanding of how mongocryptd is meant to work. 

IMO we should document the truth by the spec document and open a JAVA ticket w/ reproduction steps and requesting that they fix and always return an error. I might be wrong but I'm almost certain that mongocryptd throws an error if any document validation keywords are included in the schema.

Reply:
I opened an issue for the Java driver yesterday:
https://jira.mongodb.org/browse/JAVA-3443

Hopefully, this is enough info. If not, I can work on a gist that steps them through end-to-end.

I think Moe will get the opportunity to test the behavior with Python and Node drivers this week, so it is wording we will revisit based on those results.

Side note: just noticed another field that was getting added into the schema that doesn't include any encrypt property that I need to remove.